dev/com.ibm.ws.classloading/src/com/ibm/ws/classloading/internal/ContainerClassLoader.java

```diff
@@ -428,8 +448,18 @@ public URL getResourceURL() {
 
         @Override
         public ByteResourceInformation getByteResourceInformation() throws IOException {
-            InputStream is = this.entry.getInputStream();
-            byte[] bytes = ContainerClassLoader.getBytes(is, (int) entry.getSize());
+            byte[] bytes = null;
+            if (hook != null && resourceName.endsWith(".class")) {
+                URL resourcePath = getResourceURL();
+                if (resourcePath != null) {
+                    String className = resourceName.replace('/','.').substring(0, resourceName.length() - 6);
+                    bytes = hook.loadClass(resourcePath, className);
+                }
+            }
```

Comment:
duplicated code - lines 451-458 duplicates 250-257.  Would it make sense to put that code in a private static method - something like:
```
private byte[] getClassBytesFromHook(String resourceName, ClassLoaderHook hook) {
    byte[] bytes = null;
    if (hook != null && resourceName.endsWith(".class")) {
        URL resourcePath = getResourceURL();
        if (resourcePath != null) {
            String className = resourceName.replace('/','.').substring(0, resourceName.length() - 6);
            bytes = hook.loadClass(resourcePath, className);
        }
    }
}
```

then replace lines 451-458 and 250-257 with `byte[] bytes = getClassBytesFromHook(resourceName, hook);`

wdyt?

Reply:
I have coded up what you suggested.  Take a look.